lib/bamboo/email.ex

```diff
@@ -102,6 +117,7 @@ defmodule Bamboo.Email do
     @doc """
     Sets the #{function_name} on the email
     """
+    @spec unquote(function_name)(__MODULE__.t, String.t) :: __MODULE__.t
```

Comment:
This one is tricky so we may want to remove it for now. For example, from, to, cc, bcc accept `any` instead of `String.t`.

So maybe we should have individual functions from, to, subject, etc. with their own type specs. That may be outside the scope of this PR so maybe just removing this spec for now is ok